src/plugins/gusli/gusli_backend.cpp

```diff
@@ -0,0 +1,407 @@
+                               // cancel it so 'io' field can be free. dont care about return value
+                               // because io will get free anyways
+    }
+    nixlGusliBackendReqHSingleBdev(const nixl_xfer_op_t nixlOp,
+                                   int32_t gid,
+                                   const nixlMetaDesc &local,
+                                   const nixlMetaDesc &remote)
+        : nixlGusliBackendReqHbase(nixlOp) {
+        initCommon();
+        io.params.init_1_rng(
+            op, gid, (uint64_t)remote.addr, (uint64_t)local.len, (void *)local.addr);
+        __LOG_IO(this,
+                 ".RNG1: dev=%d, %p, 0x%lx[b], lba=0x%lx, gid=%d",
+                 remote.devId,
+                 (void *)local.addr,
+                 (uint64_t)local.len,
+                 (uint64_t)remote.addr,
+                 gid);
+    }
+    nixlGusliBackendReqHSingleBdev(const nixl_xfer_op_t nixlOp,
+                                   int32_t gid,
+                                   const nixl_meta_dlist_t &local,
+                                   const nixl_meta_dlist_t &remote)
+        : nixlGusliBackendReqHbase(nixlOp) {
+        initCommon();
+        const int num_ranges = remote.descCount();
+        gusli::io_multi_map_t *mio =
+            (gusli::io_multi_map_t *)local[0].addr; // Allocate scatter gather in the first entry
+        mio->n_entries = (num_ranges - 1); // First entry is the scatter gather
+        if (mio->my_size() > local[0].len) {
+            __LOG_ERR("mmap of sg=0x%lx[b] > is too short=0x%lx[b], Enlarge mapping or use "
+                      "shorter transfer list",
+                      mio->my_size(),
+                      local[0].len);
+            throw std::runtime_error("Failed to initialize io, See log");
+        }
+        __LOG_IO(this,
+                 ".SGL: dev=%d, %p, 0x%lx[b], lba=0x%lx, gid=%d",
+                 remote[0].devId,
+                 mio,
+                 (uint64_t)local[0].len,
+                 remote[0].addr,
+                 gid);
+        for (int i = 1; i < num_ranges; i++) { // Skip first range
+            mio->entries[i - 1] = (gusli::io_map_t){.data =
+                                                        {
+                                                            .ptr = (void *)local[i].addr,
+                                                            .byte_len = (uint64_t)local[i].len,
+                                                        },
+                                                    .offset_lba_bytes = (uint64_t)remote[i].addr};
+            __LOG_IO(this,
+                     ".RNG: dev=%d, %p, 0x%lx[b], lba=0x%lx, idx=%u",
+                     remote[i].devId,
+                     (void *)local[i].addr,
+                     (uint64_t)local[i].len,
+                     remote[i].addr,
+                     i);
+        }
+        io.params.init_multi(op, gid, *mio);
+    }
```

Comment:
Please add new-lines in after function/class definitions here and in all other places.